### ISSUE TITLE
llvm-config broke with LLVM 9.0.0-3 update, maybe other tools too

### ISSUE THREAD

**Issue by pal1000** [msys2/MINGW-packages#6121]
llvm-config broke with LLVM 9.0.0-3 update:
````
Enter MSYS2 command:/mingw32/bin/llvm-config

C:/Software/DEVELO~1/projects/mesa/msys64/mingw32/bin/llvm-config.exe: error while loading shared libraries: ?: cannot open shared object file: No such file or directory

Enter MSYS2 command:/mingw64/bin/llvm-config

C:/Software/DEVELO~1/projects/mesa/msys64/mingw64/bin/llvm-config.exe: error while loading shared libraries: libstdc++-6.dll: cannot open shared object file: No such file or directory

Enter MSYS2 command:exit
````
Victim #6120.

**Comment by mati865** [comment]
Works fine in my local env:
```
$ pacman -Q mingw-w64-x86_64-llvm
mingw-w64-x86_64-llvm 9.0.0-3


$ ntldd llvm-config.exe
        libLLVM.dll => D:\msys64\mingw64\bin\libLLVM.dll (0x0000000000db0000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x0000000005d50000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x0000000005d50000)
        libstdc++-6.dll => D:\msys64\mingw64\bin\libstdc++-6.dll (0x0000000005d50000)


$ ntldd libLLVM.dll
        zlib1.dll => D:\msys64\mingw64\bin\zlib1.dll (0x0000000000d60000)
        ADVAPI32.dll => C:\WINDOWS\SYSTEM32\ADVAPI32.dll (0x0000000000d60000)
        KERNEL32.dll => C:\WINDOWS\SYSTEM32\KERNEL32.dll (0x0000000000d80000)
        msvcrt.dll => C:\WINDOWS\SYSTEM32\msvcrt.dll (0x0000000000d80000)
        ole32.dll => C:\WINDOWS\SYSTEM32\ole32.dll (0x0000000005df0000)
        libwinpthread-1.dll => D:\msys64\mingw64\bin\libwinpthread-1.dll (0x0000000000020000)
        SHELL32.dll => C:\WINDOWS\SYSTEM32\SHELL32.dll (0x00000000067f0000)
        libgcc_s_seh-1.dll => D:\msys64\mingw64\bin\libgcc_s_seh-1.dll (0x0000000000d60000)
        libstdc++-6.dll => D:\msys64\mingw64\bin\libstdc++-6.dll (0x0000000005df0000)
        libffi-6.dll => D:\msys64\mingw64\bin\libffi-6.dll (0x0000000000020000)
        libz3.dll => D:\msys64\mingw64\bin\libz3.dll (0x00000000067f0000)
```
Seems some library is missing.

**Comment by spl1inter** [comment]
Some DLL dependencies are optional. Its different than dynamic linking in Linux, [check with dependency walker](http://www.dependencywalker.com/) and confirm.

**Comment by pal1000** [comment]
@spl1inter, considering I was able to get the CI here to replicate it in #6120 I don't think it's something specific to my system.

**Comment by spl1inter** [comment]
@pal1000 fair enough. Just to let you know I went and checked mingw-w64-x86_64-```
llvm-9.0.0-3 into a cleanroom environment on Windows 10 and had no dependency issues:
D:\Download\Temp\llvm\bin>llvm-config.exe
usage: llvm-config <OPTION>... [<COMPONENT>...]

Get various configuration information needed to compile programs which use
LLVM.  Typically called from 'configure' scripts.  Examples:
  llvm-config --cxxflags
  llvm-config --ldflags
  llvm-config --libs engine bcreader scalaropts
```

Is this perhaps something specific to Azure or the pipeline being used?

**Comment by lazka** [comment]
I get the same error locally

**Comment by lazka** [comment]
In my case it was missing `pacman -S mingw-w64-x86_64-z3`, can you try that?

**Comment by pal1000** [comment]
Yes, it depends on mingw-w64-z3. As soon as I installed it llvm-config started working.

**Comment by lazka** [comment]
ok, maybe you can add a temporary dependency in your PR as a workaround (with a comment)

**Comment by pal1000** [comment]
> ok, maybe you can add a temporary dependency in your PR as a workaround (with a comment)

Done.

**Comment by pal1000** [comment]
Fix is live in 9.0.0-5. Closing.

**Comment by lazka** [comment]
Thanks!
